app/scripts/metamask-controller.js

```diff
@@ -3316,7 +3317,11 @@ export default class MetamaskController extends EventEmitter {
     const providerNetworkState = await this.getProviderNetworkState(origin);
 
     return {
-      isUnlocked: this.isUnlocked(),
+      /**
+       * We default `isUnlocked` to `true` because even though we no longer emit events depending on this,
+       * embedded dapp providers might listen directly to our streams, and therefore depend on it, so we leave it here.
+       */
+      isUnlocked: true,
```

Comment:
a bit unfortunate in retrospect that we batched these into the same release since we are probably looking to release this as a hotfix...